README.md

```diff
@@ -1,3 +1,92 @@
+
+##	MEG preprocessing step 2
+1.	Script name will be **03_megpreprocess2**, and I would like it to do the following:
+-	Adds win/loss/neutral markers to cues
+-	Adds win/loss/neutral markers to responses
+-	Filter data based on existing filtering parameters file
+2.	Problems to figure out:
+-	When I do this one command at a time (which is how I’ve been doing this so far), it always misses two markers that I have to then add manually, so I need to figure out how to get it to add those two markers on its own.
+3.	Input: MEG file, win/loss/neutral marker files from the step above
+4.	Output: edited MEG file
+
+##	MEG processing with SAM
+1.	Script name will be **04_sam** and will do the following:
+-	orthohull (from subject’s MRI folder)
+-	Source localization in high gamma using three SAM commands and parameter file→ output is .nii file
+-	Copy .nii file from subject’s MEG directory to their MRI directory
+-	Talairach SAM results for analysis in afni
+-	Copy talairached .nii file into a group folder for group analysis
+2.	Problems:
+-	This step uses a parameter file that I would like to be able to create and distribute among subject directories whenever we want to change parameters (e.g. the time window we’re looking at in the task)
+-	Each of the three SAM commands takes a bit of time, but I heard that using swarm in biowulf may be a good way to do this. I am not sure how it would work with a loop just yet, but it’s an aspiration to make this somewhat efficient if possible.
+3.	Input: MRI with fiducial markers set, MEG file, parameter file (highgamma.param), [maybe demographics spreadsheet]
+4.	Output: .nii file, adds file path of the .nii file to a text file for analysis
+-	This text file is for 3dLME and also needs to have participant demographic info; I was manually adding this from another spreadsheet, but maybe I can have this script copy those demographic columns in and match the file path based on subject ID?
+
+##	Analysis
+1.	Right now, I will plan to make this a step for 3dLME in afni, but we may choose to do a different analysis in the coming weeks.
+2.	Script name will be **05_3dLME** and will do the following:
+-	Runs 3dLME on all the .nii files
+-	I already have a tcsh script that does this, so I’m not sure if this is really necessary to make it in python?
+3.	Input: do_3dLME_MID_highgamma2.txt (this is the text file I mentioned above that contains file paths of the .nii files and demographic info of participants, such as group and age)
+4.	Output: text file named by date and type of analysis with results from 3dLME
+
+##	Permissions
+1.	I want a script that will set all of the permissions correctly for my group for all of the output files above because umask in my bash profile does not seem to work for this.
+2.	The commands in this script may just be put at the end of each of the above ones depending on how this works out, or should it just be a bash script?
```

Comment:
Bash is fine. Python would be better. There's not too much point in rewriting lots of bash scripts into python. With that said if you tie everything together with python it will be easier to maintain and debug

Reply:
It occurred to me that some tools completely ignore umask. Perhaps that is your problem. Assuming you have written all your output to "/data/$USER/output_dir" you could make sure your output is well behaved regarding permissions by ending each swarm job with something like:

```
OUTPUT_DIR=/data/$USER/output_dir
GROUP=mygroup
find $OUTPUT_DIR -group $USER -exec chown :$GROUP {} \; -exec chmod g+rx \;
```